infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/CommonPVCStrategy.java

```diff
@@ -242,8 +251,33 @@ public void cleanup(Workspace workspace) throws InfrastructureException {
     AccountImpl account = ((WorkspaceImpl) workspace).getAccount();
     if (isPersonalAccount(account) && accountHasNoWorkspaces(account)) {
       log.debug("Deleting the common PVC: '{}',", configuredPVCName);
-      deleteCommonPVC(workspace);
-      return;
+      String namespace = infraNamespace.replaceFirst("<username>", workspace.getNamespace());
```

Comment:
can't we simply use `factory.get(workspace)` or even `workspace.getNamespace` ?

Reply:
thank you, `factory.get(workspace)` works great